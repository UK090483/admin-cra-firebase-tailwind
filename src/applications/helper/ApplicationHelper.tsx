import { IAssessmentRecord } from "assessments/types";
import { jotFormDataMaper } from "exampleJotform";
import { get } from "lodash";
import Application from "judges/views/Judge";
import TextInput from "components/Form/FormInputs/TextInput";
import ListInput from "components/Form/FormInputs/ListInput";
import MediaUpload from "components/Form/FormInputs/MediaArrayInput";

type Key = keyof ApplicationType;

type AssessmentTypeKey = keyof IAssessmentRecord;

interface ApplicationField {
  label: string;
  type: string;

  input?: JSX.Element;
}

interface ApplicationFields {
  [key: string]: ApplicationField;
}

interface ApplicationTopic {
  fields: ApplicationFields;

  assessments?: AssessmentTypeKey[];
}

interface ApplicationType {
  [propName: string]: ApplicationTopic;
}

interface GetAllFieldsResult {
  [propName: string]: ApplicationField;
}

export interface Application extends ApplicationType {}

class ApplicationC {
  application: ApplicationType;

  constructor(a: ApplicationType) {
    this.application = a;
  }

  getTopics(): string[] {
    return Object.keys(this.application).map((key) => key);
  }

  getFields(
    topic: keyof ApplicationType
  ): { key: string; field: ApplicationField }[] {
    return Object.keys(this.application[topic].fields).map((key) => {
      return { key, field: this.application[topic].fields[key] };
    });
  }

  getAllFields(): GetAllFieldsResult {
    const res: any = {};

    Object.keys(this.application).forEach((key) => {
      Object.keys(this.application[key].fields).forEach((element) => {
        res[element] = this.application[key].fields[element];
      });
    });

    return res;
  }

  getTopicLabel(topic: keyof ApplicationType): string {
    const difference = {
      BasicInformation: "Basic information",
      StartupOverview: "Startup overview",
    };

    // @ts-ignore
    return difference[topic] ? difference[topic] : topic;
  }

  getAssessments(
    topic: keyof ApplicationType
  ): AssessmentTypeKey[] | undefined {
    return this.application[topic].assessments;
  }

  parseJotFormData(jotformData: IJotFormData) {
    const res: IJotFormData = {};
    Object.keys(this.application).forEach((key) => {
      Object.keys(this.application[key].fields).forEach((element) => {
        res[element] = get(jotformData, jotFormDataMaper[element]);
      });
    });

    Object.keys(jotFormDataMaper).forEach((element) => {
      res[element] = get(jotformData, jotFormDataMaper[element]);
    });

    return res;
  }
}

interface IJotFormData {
  [key: string]: any;
}

const ApplicationObject: ApplicationType = {
  BasicInformation: {
    fields: {
      startupName: {
        label: "Startup Name",
        type: "text",
        input: <TextInput label={"Startup Name"} name={"startupName"} />,
      },
      tagLine: {
        label: "Tag Line",
        type: "text",
        input: <TextInput label={"Tag Line"} name={"tagLine"} />,
      },
      website: {
        label: "Website",
        type: "url",
        input: <TextInput label={"Website"} name={"website"} />,
      },
      foundingDate: {
        label: "Founding date",
        type: "text",
        input: <TextInput label={"Founding date"} name={"foundingDate"} />,
      },
      stage: {
        label: "Stage",
        type: "text",
        input: <TextInput label={"Stage"} name={"stage"} />,
      },
      segmentFocus: {
        label: "Segment focus",
        type: "text",
        input: <TextInput label={"Segment focus"} name={"segmentFocus"} />,
      },
      industry: {
        label: "Industry",
        type: "text",
        input: <TextInput label={"Industry"} name={"industry"} />,
      },
      headquarters: {
        label: "Headquarters",
        type: "text",
        input: <TextInput label={"Headquarters"} name={"headquarters"} />,
      },
    },
  },
  StartupOverview: {
    fields: {
      startupOverview: {
        label: "Startup overview",
        type: "text",
        input: (
          <TextInput
            label={"startup Overview"}
            name={"startupOverview"}
            textarea
          />
        ),
      },
    },
  },
  Product: {
    fields: {
      problemYouAreSolving: {
        label: "Problem you are solving & severity of the problem",
        type: "text",
        input: (
          <TextInput
            label={"Problem you are solving & severity of the problem"}
            name={"problemYouAreSolving"}
            textarea
          />
        ),
      },
      productDescription: {
        label: "Product description",
        type: "text",
        input: (
          <TextInput
            label={"Product description"}
            name={"productDescription"}
            textarea
          />
        ),
      },
      whatMakesYourProductUnique: {
        label: "What makes your service/product special and unique?",
        type: "text",
        input: (
          <TextInput
            label={"What makes your service/product special and unique?"}
            name={"whatMakesYourProductUnique"}
            textarea
          />
        ),
      },
      productReadiness: {
        label: "Product readiness & where are you now",
        type: "text",
        input: (
          <TextInput
            label={"Product readiness & where are you now"}
            name={"productReadiness"}
            textarea
          />
        ),
      },
      productRoadMap: {
        label: "Product Road Map",
        type: "text",
        input: (
          <TextInput
            label={"Product Road Map"}
            name={"productRoadMap"}
            textarea
          />
        ),
      },
    },
    assessments: ["isInnovative", "isClearUSP"],
  },
  Market: {
    fields: {
      targetMarket: {
        label: "Target Market",
        type: "list",
        input: <ListInput label={"Target Market"} name={"targetMarket"} />,
      },
      marketDrivers: {
        label: "Market Drivers & challenges",
        type: "text",
        input: (
          <TextInput
            label={"Market Drivers & challenges"}
            name={"marketDrivers"}
            textarea
          />
        ),
      },
      marketSize: {
        label: "Market Size",
        type: "text",
        input: <TextInput label={"Market Size"} name={"marketSize"} textarea />,
      },
    },

    assessments: ["isTherePotential"],
  },
  Competition: {
    fields: {
      competition: {
        label: "Competition",
        type: "text",
        input: (
          <TextInput label={"Competition"} name={"competition"} textarea />
        ),
      },
      strengthAgainstCompetitors: {
        label: "Strength against competitors",
        type: "text",
        input: (
          <TextInput
            label={"Strength against competitors"}
            name={"strengthAgainstCompetitors"}
            textarea
          />
        ),
      },
      challengesAgainstCompetitors: {
        label: "Challenges against competitors",
        type: "text",
        input: (
          <TextInput
            label={"Challenges against competitors"}
            name={"challengesAgainstCompetitors"}
            textarea
          />
        ),
      },
    },

    assessments: ["doTheyKnowCompetitors"],
  },
  Business: {
    fields: {
      revenueSources: {
        label: "Revenue sources",
        type: "text",
        input: (
          <TextInput
            label={"Revenue sources"}
            name={"revenueSources"}
            textarea
          />
        ),
      },
      salesStrategy: {
        label: "Sales strategy & Pricing model",
        type: "text",
        input: (
          <TextInput
            label={"Sales strategy & Pricing model"}
            name={"salesStrategy"}
            textarea
          />
        ),
      },
      traction: {
        label: "Traction",
        type: "text",
        input: <TextInput label={"traction"} name={"traction"} textarea />,
      },
      pilotsCustomers: {
        label: "Pilots & customers",
        type: "text",
        input: (
          <TextInput
            label={"Pilots & customers"}
            name={"pilotsCustomers"}
            textarea
          />
        ),
      },
      metrics: {
        label: "Metrics",
        type: "text",
        input: <TextInput label={"Metrics"} name={"metrics"} textarea />,
      },
    },

    assessments: ["isBusinessModelSolid", "areNumbersPromising"],
  },
  Funding: {
    fields: {
      howIsCompanyFunded: {
        label: "How is the company funded?",
        type: "text",
        input: (
          <TextInput
            label={"How is the company funded?"}
            name={"howIsCompanyFunded"}
            textarea
          />
        ),
      },
      amountRaisedToDate: {
        label: "Amount raised to date",
        type: "text",
        input: (
          <TextInput
            label={"Amount raised to date"}
            name={"amountRaisedToDate"}
            textarea
          />
        ),
      },
      yourInvestors: {
        label: "Your investors",
        type: "text",
        input: (
          <TextInput label={"Your investors"} name={"yourInvestors"} textarea />
        ),
      },
      fundingObjective: {
        label: "Funding objective",
        type: "text",
        input: (
          <TextInput
            label={"Funding objective"}
            name={"fundingObjective"}
            textarea
          />
        ),
      },
    },

    assessments: ["isFundingSolid"],
  },
  Team: {
    fields: {
      numberOfFounders: {
        label: "Number of founders",
        type: "text",
        input: (
          <TextInput label={"Number of founders"} name={"numberOfFounders"} />
        ),
      },
      overallTeamSize: {
        label: "Overall team size (including founders)",
        type: "text",
        input: (
          <TextInput
            label={"Overall team size (including founders)"}
            name={"overallTeamSize"}
          />
        ),
      },
      foundersQualification: {
        label: "Founder’s qualification/ background",
        type: "text",
        input: (
          <TextInput
            label={"Founder’s qualification/ background"}
            name={"foundersQualification"}
            textarea
          />
        ),
      },
    },

    assessments: ["isTeamExperienced"],
  },
  Impact: {
    fields: {
      howImpact: {
        label:
          "How can your startup help generate an impact for our future cities, especially Hamburg ?",
        type: "text",
        input: (
          <TextInput
            label={
              "How can your startup help generate an impact for our future cities, especially Hamburg ?"
            }
            name={"howImpact"}
            textarea
          />
        ),
      },
      whatDegreeSustainability: {
        label:
          "To what degree is sustainability integrated into the business model ?",
        type: "text",
        input: (
          <TextInput
            label={
              "To what degree is sustainability integrated into the business model ?"
            }
            name={"whatDegreeSustainability"}
            textarea
          />
        ),
      },
      connectionToHamburg: {
        label: "Do you have a connection to Hamburg ?",
        type: "text",
        input: (
          <TextInput
            label={"Do you have a connection to Hamburg ?"}
            name={"connectionToHamburg"}
            textarea
          />
        ),
      },
      collaborateWithEntityHamburg: {
        label: "Do you already collaborate with any entity from Hamburg ?",
        type: "text",
        input: (
          <TextInput
            label={"Do you already collaborate with any entity from Hamburg ?"}
            name={"collaborateWithEntityHamburg"}
            textarea
          />
        ),
      },
      howDidYouHearAboutFHA: {
        label: "How did you hear about the Future Hamburg Award ?",
        type: "text",
        input: (
          <TextInput
            label={"How did you hear about the Future Hamburg Award ?"}
            name={"howDidYouHearAboutFHA"}
            textarea
          />
        ),
      },
    },

    assessments: ["willFutureCitiesBenefit", "isComplianceWithSDG"],
  },
  Media: {
    fields: {
      website: {
        label: "Website",
        type: "text",
        input: <TextInput label={"Website"} name={"website"} />,
      },
      companyLogo: {
        label: "Company Logo",
        type: "mediaArray",
        input: <MediaUpload label={"Company Logo"} name={"companyLogo"} />,
      },
      companyDeck: {
        label: "Company Deck",
        type: "mediaArray",
        input: <MediaUpload label={"Company Deck"} name={"companyDeck"} />,
      },
    },

    assessments: ["isPresentationGood"],
  },
};

const ApplicationHelper = new ApplicationC(ApplicationObject);
export default Application;

export type { ApplicationType, Key };
export { ApplicationHelper };
