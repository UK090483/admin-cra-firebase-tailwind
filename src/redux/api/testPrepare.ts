//  @jest-environment node
import { db, firebase } from "misc/firebase";

const buildTestCollection = async () => {
  const testArray = new Array(10);
  testArray.fill(1);
  return Promise.all(
    testArray.map((item, index) =>
      db
        .collection("testCollection")
        .doc("item" + index)
        .set({
          name: index,
          updated_at: firebase.firestore.Timestamp.now().toDate(),
        })
    )
  );
};

interface item {
  id: string;
}

const deleteTestCollection = async () => {
  const erasers: any[] = [];
  try {
    const collection = await db.collection("testCollection").get();
    collection.forEach((item: item) => {
      erasers.push(db.collection("testCollection").doc(item.id).delete());
    });
  } catch (error) {
    console.log(error);
  }

  return true;
  return Promise.all(erasers);
};

export { buildTestCollection, deleteTestCollection, db };
