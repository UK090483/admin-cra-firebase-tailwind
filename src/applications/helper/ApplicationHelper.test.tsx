/**
 * @jest-environment jsdom
 */
import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { ApplicationHelper } from "./ApplicationHelper";
import { TestWrap } from "tests/testPrepare";

describe("ApplicationHelper", () => {
  it("Match getTopics", async () => {
    expect(ApplicationHelper.getTopics()).toMatchSnapshot();
  });

  it("Match getFields", async () => {
    ApplicationHelper.getTopics().forEach((topic) => {
      expect(ApplicationHelper.getFields(topic)).toMatchSnapshot();
    });
  });

  it("Match getAllFields", async () => {
    expect(ApplicationHelper.getAllFields()).toMatchSnapshot();
  });

  it("Match getAssessments", async () => {
    ApplicationHelper.getTopics().forEach((topic) => {
      expect(ApplicationHelper.getAssessments(topic)).toMatchSnapshot();
    });
  });
  it("Match parseJotFormData", async () => {
    expect(ApplicationHelper.parseJotFormData({})).toMatchSnapshot();
  });
});
