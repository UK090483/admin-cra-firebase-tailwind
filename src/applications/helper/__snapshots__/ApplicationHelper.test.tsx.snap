// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationHelper Match getAllFields 1`] = `
Object {
  "amountRaisedToDate": Object {
    "input": <TextInput
      label="Amount raised to date"
      name="amountRaisedToDate"
      textarea={true}
    />,
    "label": "Amount raised to date",
    "type": "text",
  },
  "challengesAgainstCompetitors": Object {
    "input": <TextInput
      label="Challenges against competitors"
      name="challengesAgainstCompetitors"
      textarea={true}
    />,
    "label": "Challenges against competitors",
    "type": "text",
  },
  "collaborateWithEntityHamburg": Object {
    "input": <TextInput
      label="Do you already collaborate with any entity from Hamburg ?"
      name="collaborateWithEntityHamburg"
      textarea={true}
    />,
    "label": "Do you already collaborate with any entity from Hamburg ?",
    "type": "text",
  },
  "companyDeck": Object {
    "input": <MediaUpload
      label="Company Deck"
      name="companyDeck"
    />,
    "label": "Company Deck",
    "type": "mediaArray",
  },
  "companyLogo": Object {
    "input": <MediaUpload
      label="Company Logo"
      name="companyLogo"
    />,
    "label": "Company Logo",
    "type": "mediaArray",
  },
  "competition": Object {
    "input": <TextInput
      label="Competition"
      name="competition"
      textarea={true}
    />,
    "label": "Competition",
    "type": "text",
  },
  "connectionToHamburg": Object {
    "input": <TextInput
      label="Do you have a connection to Hamburg ?"
      name="connectionToHamburg"
      textarea={true}
    />,
    "label": "Do you have a connection to Hamburg ?",
    "type": "text",
  },
  "foundersQualification": Object {
    "input": <TextInput
      label="Founder’s qualification/ background"
      name="foundersQualification"
      textarea={true}
    />,
    "label": "Founder’s qualification/ background",
    "type": "text",
  },
  "foundingDate": Object {
    "input": <TextInput
      label="Founding date"
      name="foundingDate"
    />,
    "label": "Founding date",
    "type": "text",
  },
  "fundingObjective": Object {
    "input": <TextInput
      label="Funding objective"
      name="fundingObjective"
      textarea={true}
    />,
    "label": "Funding objective",
    "type": "text",
  },
  "headquarters": Object {
    "input": <TextInput
      label="Headquarters"
      name="headquarters"
    />,
    "label": "Headquarters",
    "type": "text",
  },
  "howDidYouHearAboutFHA": Object {
    "input": <TextInput
      label="How did you hear about the Future Hamburg Award ?"
      name="howDidYouHearAboutFHA"
      textarea={true}
    />,
    "label": "How did you hear about the Future Hamburg Award ?",
    "type": "text",
  },
  "howImpact": Object {
    "input": <TextInput
      label="How can your startup help generate an impact for our future cities, especially Hamburg ?"
      name="howImpact"
      textarea={true}
    />,
    "label": "How can your startup help generate an impact for our future cities, especially Hamburg ?",
    "type": "text",
  },
  "howIsCompanyFunded": Object {
    "input": <TextInput
      label="How is the company funded?"
      name="howIsCompanyFunded"
      textarea={true}
    />,
    "label": "How is the company funded?",
    "type": "text",
  },
  "industry": Object {
    "input": <TextInput
      label="Industry"
      name="industry"
    />,
    "label": "Industry",
    "type": "text",
  },
  "marketDrivers": Object {
    "input": <TextInput
      label="Market Drivers & challenges"
      name="marketDrivers"
      textarea={true}
    />,
    "label": "Market Drivers & challenges",
    "type": "text",
  },
  "marketSize": Object {
    "input": <TextInput
      label="Market Size"
      name="marketSize"
      textarea={true}
    />,
    "label": "Market Size",
    "type": "text",
  },
  "metrics": Object {
    "input": <TextInput
      label="Metrics"
      name="metrics"
      textarea={true}
    />,
    "label": "Metrics",
    "type": "text",
  },
  "numberOfFounders": Object {
    "input": <TextInput
      label="Number of founders"
      name="numberOfFounders"
    />,
    "label": "Number of founders",
    "type": "text",
  },
  "overallTeamSize": Object {
    "input": <TextInput
      label="Overall team size (including founders)"
      name="overallTeamSize"
    />,
    "label": "Overall team size (including founders)",
    "type": "text",
  },
  "pilotsCustomers": Object {
    "input": <TextInput
      label="Pilots & customers"
      name="pilotsCustomers"
      textarea={true}
    />,
    "label": "Pilots & customers",
    "type": "text",
  },
  "problemYouAreSolving": Object {
    "input": <TextInput
      label="Problem you are solving & severity of the problem"
      name="problemYouAreSolving"
      textarea={true}
    />,
    "label": "Problem you are solving & severity of the problem",
    "type": "text",
  },
  "productDescription": Object {
    "input": <TextInput
      label="Product description"
      name="productDescription"
      textarea={true}
    />,
    "label": "Product description",
    "type": "text",
  },
  "productReadiness": Object {
    "input": <TextInput
      label="Product readiness & where are you now"
      name="productReadiness"
      textarea={true}
    />,
    "label": "Product readiness & where are you now",
    "type": "text",
  },
  "productRoadMap": Object {
    "input": <TextInput
      label="Product Road Map"
      name="productRoadMap"
      textarea={true}
    />,
    "label": "Product Road Map",
    "type": "text",
  },
  "revenueSources": Object {
    "input": <TextInput
      label="Revenue sources"
      name="revenueSources"
      textarea={true}
    />,
    "label": "Revenue sources",
    "type": "text",
  },
  "salesStrategy": Object {
    "input": <TextInput
      label="Sales strategy & Pricing model"
      name="salesStrategy"
      textarea={true}
    />,
    "label": "Sales strategy & Pricing model",
    "type": "text",
  },
  "segmentFocus": Object {
    "input": <TextInput
      label="Segment focus"
      name="segmentFocus"
    />,
    "label": "Segment focus",
    "type": "text",
  },
  "stage": Object {
    "input": <TextInput
      label="Stage"
      name="stage"
    />,
    "label": "Stage",
    "type": "text",
  },
  "startupName": Object {
    "input": <TextInput
      label="Startup Name"
      name="startupName"
    />,
    "label": "Startup Name",
    "type": "text",
  },
  "startupOverview": Object {
    "input": <TextInput
      label="startup Overview"
      name="startupOverview"
      textarea={true}
    />,
    "label": "Startup overview",
    "type": "text",
  },
  "strengthAgainstCompetitors": Object {
    "input": <TextInput
      label="Strength against competitors"
      name="strengthAgainstCompetitors"
      textarea={true}
    />,
    "label": "Strength against competitors",
    "type": "text",
  },
  "tagLine": Object {
    "input": <TextInput
      label="Tag Line"
      name="tagLine"
    />,
    "label": "Tag Line",
    "type": "text",
  },
  "targetMarket": Object {
    "input": <ListInput
      label="Target Market"
      name="targetMarket"
    />,
    "label": "Target Market",
    "type": "list",
  },
  "traction": Object {
    "input": <TextInput
      label="traction"
      name="traction"
      textarea={true}
    />,
    "label": "Traction",
    "type": "text",
  },
  "website": Object {
    "input": <TextInput
      label="Website"
      name="website"
    />,
    "label": "Website",
    "type": "text",
  },
  "whatDegreeSustainability": Object {
    "input": <TextInput
      label="To what degree is sustainability integrated into the business model ?"
      name="whatDegreeSustainability"
      textarea={true}
    />,
    "label": "To what degree is sustainability integrated into the business model ?",
    "type": "text",
  },
  "whatMakesYourProductUnique": Object {
    "input": <TextInput
      label="What makes your service/product special and unique?"
      name="whatMakesYourProductUnique"
      textarea={true}
    />,
    "label": "What makes your service/product special and unique?",
    "type": "text",
  },
  "yourInvestors": Object {
    "input": <TextInput
      label="Your investors"
      name="yourInvestors"
      textarea={true}
    />,
    "label": "Your investors",
    "type": "text",
  },
}
`;

exports[`ApplicationHelper Match getAssessments 1`] = `undefined`;

exports[`ApplicationHelper Match getAssessments 2`] = `undefined`;

exports[`ApplicationHelper Match getAssessments 3`] = `
Array [
  "isInnovative",
  "isClearUSP",
]
`;

exports[`ApplicationHelper Match getAssessments 4`] = `
Array [
  "isTherePotential",
]
`;

exports[`ApplicationHelper Match getAssessments 5`] = `
Array [
  "doTheyKnowCompetitors",
]
`;

exports[`ApplicationHelper Match getAssessments 6`] = `
Array [
  "isBusinessModelSolid",
  "areNumbersPromising",
]
`;

exports[`ApplicationHelper Match getAssessments 7`] = `
Array [
  "isFundingSolid",
]
`;

exports[`ApplicationHelper Match getAssessments 8`] = `
Array [
  "isTeamExperienced",
]
`;

exports[`ApplicationHelper Match getAssessments 9`] = `
Array [
  "willFutureCitiesBenefit",
  "isComplianceWithSDG",
]
`;

exports[`ApplicationHelper Match getAssessments 10`] = `
Array [
  "isPresentationGood",
]
`;

exports[`ApplicationHelper Match getFields 1`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Startup Name"
        name="startupName"
      />,
      "label": "Startup Name",
      "type": "text",
    },
    "key": "startupName",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Tag Line"
        name="tagLine"
      />,
      "label": "Tag Line",
      "type": "text",
    },
    "key": "tagLine",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Website"
        name="website"
      />,
      "label": "Website",
      "type": "url",
    },
    "key": "website",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Founding date"
        name="foundingDate"
      />,
      "label": "Founding date",
      "type": "text",
    },
    "key": "foundingDate",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Stage"
        name="stage"
      />,
      "label": "Stage",
      "type": "text",
    },
    "key": "stage",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Segment focus"
        name="segmentFocus"
      />,
      "label": "Segment focus",
      "type": "text",
    },
    "key": "segmentFocus",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Industry"
        name="industry"
      />,
      "label": "Industry",
      "type": "text",
    },
    "key": "industry",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Headquarters"
        name="headquarters"
      />,
      "label": "Headquarters",
      "type": "text",
    },
    "key": "headquarters",
  },
]
`;

exports[`ApplicationHelper Match getFields 2`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="startup Overview"
        name="startupOverview"
        textarea={true}
      />,
      "label": "Startup overview",
      "type": "text",
    },
    "key": "startupOverview",
  },
]
`;

exports[`ApplicationHelper Match getFields 3`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Problem you are solving & severity of the problem"
        name="problemYouAreSolving"
        textarea={true}
      />,
      "label": "Problem you are solving & severity of the problem",
      "type": "text",
    },
    "key": "problemYouAreSolving",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Product description"
        name="productDescription"
        textarea={true}
      />,
      "label": "Product description",
      "type": "text",
    },
    "key": "productDescription",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="What makes your service/product special and unique?"
        name="whatMakesYourProductUnique"
        textarea={true}
      />,
      "label": "What makes your service/product special and unique?",
      "type": "text",
    },
    "key": "whatMakesYourProductUnique",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Product readiness & where are you now"
        name="productReadiness"
        textarea={true}
      />,
      "label": "Product readiness & where are you now",
      "type": "text",
    },
    "key": "productReadiness",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Product Road Map"
        name="productRoadMap"
        textarea={true}
      />,
      "label": "Product Road Map",
      "type": "text",
    },
    "key": "productRoadMap",
  },
]
`;

exports[`ApplicationHelper Match getFields 4`] = `
Array [
  Object {
    "field": Object {
      "input": <ListInput
        label="Target Market"
        name="targetMarket"
      />,
      "label": "Target Market",
      "type": "list",
    },
    "key": "targetMarket",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Market Drivers & challenges"
        name="marketDrivers"
        textarea={true}
      />,
      "label": "Market Drivers & challenges",
      "type": "text",
    },
    "key": "marketDrivers",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Market Size"
        name="marketSize"
        textarea={true}
      />,
      "label": "Market Size",
      "type": "text",
    },
    "key": "marketSize",
  },
]
`;

exports[`ApplicationHelper Match getFields 5`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Competition"
        name="competition"
        textarea={true}
      />,
      "label": "Competition",
      "type": "text",
    },
    "key": "competition",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Strength against competitors"
        name="strengthAgainstCompetitors"
        textarea={true}
      />,
      "label": "Strength against competitors",
      "type": "text",
    },
    "key": "strengthAgainstCompetitors",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Challenges against competitors"
        name="challengesAgainstCompetitors"
        textarea={true}
      />,
      "label": "Challenges against competitors",
      "type": "text",
    },
    "key": "challengesAgainstCompetitors",
  },
]
`;

exports[`ApplicationHelper Match getFields 6`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Revenue sources"
        name="revenueSources"
        textarea={true}
      />,
      "label": "Revenue sources",
      "type": "text",
    },
    "key": "revenueSources",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Sales strategy & Pricing model"
        name="salesStrategy"
        textarea={true}
      />,
      "label": "Sales strategy & Pricing model",
      "type": "text",
    },
    "key": "salesStrategy",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="traction"
        name="traction"
        textarea={true}
      />,
      "label": "Traction",
      "type": "text",
    },
    "key": "traction",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Pilots & customers"
        name="pilotsCustomers"
        textarea={true}
      />,
      "label": "Pilots & customers",
      "type": "text",
    },
    "key": "pilotsCustomers",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Metrics"
        name="metrics"
        textarea={true}
      />,
      "label": "Metrics",
      "type": "text",
    },
    "key": "metrics",
  },
]
`;

exports[`ApplicationHelper Match getFields 7`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="How is the company funded?"
        name="howIsCompanyFunded"
        textarea={true}
      />,
      "label": "How is the company funded?",
      "type": "text",
    },
    "key": "howIsCompanyFunded",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Amount raised to date"
        name="amountRaisedToDate"
        textarea={true}
      />,
      "label": "Amount raised to date",
      "type": "text",
    },
    "key": "amountRaisedToDate",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Your investors"
        name="yourInvestors"
        textarea={true}
      />,
      "label": "Your investors",
      "type": "text",
    },
    "key": "yourInvestors",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Funding objective"
        name="fundingObjective"
        textarea={true}
      />,
      "label": "Funding objective",
      "type": "text",
    },
    "key": "fundingObjective",
  },
]
`;

exports[`ApplicationHelper Match getFields 8`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Number of founders"
        name="numberOfFounders"
      />,
      "label": "Number of founders",
      "type": "text",
    },
    "key": "numberOfFounders",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Overall team size (including founders)"
        name="overallTeamSize"
      />,
      "label": "Overall team size (including founders)",
      "type": "text",
    },
    "key": "overallTeamSize",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Founder’s qualification/ background"
        name="foundersQualification"
        textarea={true}
      />,
      "label": "Founder’s qualification/ background",
      "type": "text",
    },
    "key": "foundersQualification",
  },
]
`;

exports[`ApplicationHelper Match getFields 9`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="How can your startup help generate an impact for our future cities, especially Hamburg ?"
        name="howImpact"
        textarea={true}
      />,
      "label": "How can your startup help generate an impact for our future cities, especially Hamburg ?",
      "type": "text",
    },
    "key": "howImpact",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="To what degree is sustainability integrated into the business model ?"
        name="whatDegreeSustainability"
        textarea={true}
      />,
      "label": "To what degree is sustainability integrated into the business model ?",
      "type": "text",
    },
    "key": "whatDegreeSustainability",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Do you have a connection to Hamburg ?"
        name="connectionToHamburg"
        textarea={true}
      />,
      "label": "Do you have a connection to Hamburg ?",
      "type": "text",
    },
    "key": "connectionToHamburg",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="Do you already collaborate with any entity from Hamburg ?"
        name="collaborateWithEntityHamburg"
        textarea={true}
      />,
      "label": "Do you already collaborate with any entity from Hamburg ?",
      "type": "text",
    },
    "key": "collaborateWithEntityHamburg",
  },
  Object {
    "field": Object {
      "input": <TextInput
        label="How did you hear about the Future Hamburg Award ?"
        name="howDidYouHearAboutFHA"
        textarea={true}
      />,
      "label": "How did you hear about the Future Hamburg Award ?",
      "type": "text",
    },
    "key": "howDidYouHearAboutFHA",
  },
]
`;

exports[`ApplicationHelper Match getFields 10`] = `
Array [
  Object {
    "field": Object {
      "input": <TextInput
        label="Website"
        name="website"
      />,
      "label": "Website",
      "type": "text",
    },
    "key": "website",
  },
  Object {
    "field": Object {
      "input": <MediaUpload
        label="Company Logo"
        name="companyLogo"
      />,
      "label": "Company Logo",
      "type": "mediaArray",
    },
    "key": "companyLogo",
  },
  Object {
    "field": Object {
      "input": <MediaUpload
        label="Company Deck"
        name="companyDeck"
      />,
      "label": "Company Deck",
      "type": "mediaArray",
    },
    "key": "companyDeck",
  },
]
`;

exports[`ApplicationHelper Match getTopics 1`] = `
Array [
  "BasicInformation",
  "StartupOverview",
  "Product",
  "Market",
  "Competition",
  "Business",
  "Funding",
  "Team",
  "Impact",
  "Media",
]
`;

exports[`ApplicationHelper Match parseJotFormData 1`] = `
Object {
  "amountRaisedToDate": undefined,
  "challengesAgainstCompetitors": undefined,
  "collaborateWithEntityHamburg": undefined,
  "companyDeck": undefined,
  "companyLogo": undefined,
  "competition": undefined,
  "connectionToHamburg": undefined,
  "contactEmail": undefined,
  "contactName": undefined,
  "contactPhoneNumber": undefined,
  "contactPosition": undefined,
  "foundersQualification": undefined,
  "foundingDate": undefined,
  "fundingObjective": undefined,
  "headquarters": undefined,
  "howDidYouHearAboutFHA": undefined,
  "howImpact": undefined,
  "howIsCompanyFunded": undefined,
  "id": undefined,
  "industry": undefined,
  "marketDrivers": undefined,
  "marketSize": undefined,
  "metrics": undefined,
  "numberOfFounders": undefined,
  "overallTeamSize": undefined,
  "pilotsCustomers": undefined,
  "problemYouAreSolving": undefined,
  "productDescription": undefined,
  "productReadiness": undefined,
  "productRoadMap": undefined,
  "revenueSources": undefined,
  "salesStrategy": undefined,
  "segmentFocus": undefined,
  "stage": undefined,
  "startupName": undefined,
  "startupOverview": undefined,
  "strengthAgainstCompetitors": undefined,
  "tagLine": undefined,
  "targetMarket": undefined,
  "traction": undefined,
  "website": undefined,
  "whatDegreeSustainability": undefined,
  "whatMakesYourProductUnique": undefined,
  "yourInvestors": undefined,
}
`;
